src/pybind/mgr/cephadm/module.py

```diff
@@ -3075,9 +3076,23 @@ def set_custom_prometheus_alerts(self, alerts_file: str) -> str:
 
     @handle_orch_error
     def set_prometheus_target(self, url: str) -> str:
-        valid_url_pattern = r"^(?!http:\/\/)(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}:\d{1,5})$"
-        if re.match(valid_url_pattern, url) is None:
+        if url.startswith("http://") or url.startswith("https://"):
             return f"Invalid URL '{url}'. It should be in the format host_ip:port"
+
+        try:
+            parsed_url_with_scheme = urlparse(f'http://{url}')
+            host = parsed_url_with_scheme.hostname
+            port = parsed_url_with_scheme.port
+        except ValueError as e:
+            return f"Invalid URL. {e}"
+
+        if not host or not port:
+            return f"Invalid URL '{url}'. Hostname or port is missing."
+
+        try:
+            ipaddress.ip_address(host)
+        except (ValueError, OSError) as e:
+            return f"Invalid URL. {e}"
```

Comment:
NIT: I feel like all of this could be combined into a single `try/catch` block

Reply:
Thanks, updated the PR